src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.cs

```diff
@@ -424,5 +427,41 @@ internal string GetRealAnalyzerLoadPath(string originalFullPath)
             }
             return null;
         }
+
+        internal string? RedirectAssemblyPathExternally(string fullPath)
+        {
+            CheckIfDisposed();
+
+            if (!_externalRedirectors.IsDefaultOrEmpty)
+            {
+                string? redirectedPath = null;
+
+                foreach (IAnalyzerAssemblyRedirector redirector in _externalRedirectors)
+                {
+                    try
+                    {
+                        if (redirector.RedirectPath(fullPath) is { } currentlyRedirectedPath)
+                        {
+                            if (redirectedPath == null)
+                            {
+                                redirectedPath = currentlyRedirectedPath;
+                            }
+                            else if (redirectedPath != currentlyRedirectedPath)
+                            {
+                                throw new InvalidOperationException($"Multiple redirectors disagree on the path to redirect '{fullPath}' to ('{redirectedPath}' vs '{currentlyRedirectedPath}').");
```

Comment:
The `catch` below means this exception never manifests correct? It's just a Watson report mechanism?

Reply:
Yes, we don't want to continue if there's a mismatch, but we probably shouldn't crash either.